code/modules/projectiles/gun_helpers.dm

```diff
@@ -518,10 +518,10 @@ DEFINES in setup.dm, referenced here.
 
 	if(istype(slot) && (slot.storage_flags & STORAGE_ALLOW_QUICKDRAW))
 		for(var/obj/cycled_weapon in slot.return_inv())
-			if(isweapon(cycled_weapon))
+			if(isweapon(cycled_weapon) || cycled_weapon.flags_atom & QUICK_DRAWABLE)
```

Comment:
Honestly just remove the weapon check and slap the quickdraw flag on all weapons

Reply:
yea just thought of that